lib/core/extension/vote_tally_extension.dart

```diff
@@ -0,0 +1,25 @@
+extension VoteTallyExtension on Map<String, dynamic> {
+  Map<String, dynamic> calculateUnityAndQuorum() {
+    if (this['votetally'] != null && this['votetally'].isNotEmpty) {
+      double supply = 0;
+      final double abstain = this['votetally'][0]['abstain_votePower_a'].quantityAsDouble();
+      final double pass = this['votetally'][0]['pass_votePower_a'].quantityAsDouble();
+      final double fail = this['votetally'][0]['fail_votePower_a'].quantityAsDouble();
+      final double unity = (pass + fail > 0) ? pass / (pass + fail) : 0;
+
+      if (this['details_ballotSupply_a'] != null) {
```

Comment:
one more... 

double supply = this['details_ballotSupply_a']?.quantityAsDouble() ?? 0;